cmd/minikube/cmd/start_test.go

```diff
@@ -126,6 +127,71 @@ func TestStartClusterUpWithFlag(t *testing.T) {
 	assertCommandLineArguments(expectedArguments, t)
 }
 
+func TestShellProxyVariableWithProxyFlag(t *testing.T) {
+	setUp(t)
+	defer os.RemoveAll(testDir)
+	defer minitesting.ResetDefaultRoundTripper()
+	defer SetRunner(util.RealRunner{})
+	defer viper.Reset()
+
+	expectedShellProxyVariable := "http_proxy=http://localhost:3128 https_proxy=https://localhost:3128 no_proxy=localhost,127.0.0.1"
+
+	viper.Set("http-proxy", "http://localhost:3128")
+	viper.Set("https-proxy", "https://localhost:3128")
+	viper.Set("no-proxy", "localhost,127.0.0.1")
+
+	setProxy()
+
+	if shellProxyEnv != expectedShellProxyVariable {
+		t.Fatalf("Shell variable doesn't set properly expected %s got %s", expectedShellProxyVariable, shellProxyEnv)
+	}
+}
+
+func TestDockerEnvWithProxyFlag(t *testing.T) {
+	setUp(t)
+	defer os.RemoveAll(testDir)
+	defer minitesting.ResetDefaultRoundTripper()
+	defer SetRunner(util.RealRunner{})
+	defer viper.Reset()
+
+	expectedDockerEnv := []string{"HTTP_PROXY=http://localhost:3128",
+		"HTTPS_PROXY=https://localhost:3128",
+		fmt.Sprintf("NO_PROXY=%s,10.0.0.1", updateNoProxyForDocker())}
+
+	viper.Set("http-proxy", "http://localhost:3128")
+	viper.Set("https-proxy", "https://localhost:3128")
+	viper.Set("no-proxy", "10.0.0.1")
+
+	setProxy()
+
+	assertCompareSlice(expectedDockerEnv, dockerEnv, t)
+}
+
+func TestClusterUpWithProxyFlag(t *testing.T) {
+	setUp(t)
+	defer os.RemoveAll(testDir)
+	defer minitesting.ResetDefaultRoundTripper()
+	defer SetRunner(util.RealRunner{})
+	defer viper.Reset()
+
+	viper.Set("http-proxy", "http://localhost:3128")
```

Comment:
See, here you need to tests flags and proxy configuration code together. If you separate this you can so the Viper/command line thing independent from the configuration part. 

Otherwise good step forward with the tests